File: algorithms/2_medium/C/03577_count_the_number_of_computer_unlocking_permutations/solution.py
MOD_TO = 1_000_000_007

class Solution:
    def countPermutations(self, complexity: List[int]) -> int:
        n = len(complexity)

        for i in range(1, n):
            if complexity[i] <= complexity[0]:
                return 0

        count = 1

        for i in range(2, n):
            count = (count * i) % MOD_TO

        return count

File: algorithms/2_medium/C/03577_count_the_number_of_computer_unlocking_permutations/solution.rs
impl Solution {
    const MOD_TO: i64 = 1_000_000_007;

    pub fn count_permutations(complexity: Vec<i32>) -> i32 {
        let len = complexity.len();

        for i in 1..len {
            if complexity[i] <= complexity[0] {
                return 0;
            }
        }

        let mut count = 1_i64;

        for i in 2..(len as i64) {
            count = (count * i) % Self::MOD_TO;
        }

        count as i32
    }
}
